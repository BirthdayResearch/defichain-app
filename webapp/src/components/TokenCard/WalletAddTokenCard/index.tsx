import React from 'react';
import { Card, CardBody, Col, Row, Button } from 'reactstrap';
import { I18n } from 'react-redux-i18n';

import styles from '../TokenCard.module.scss';
import { IWalletTokenCard } from '../../../utils/interfaces';
// import Icon from '../../../assets/svg/icon-coin-bitcoin-lapis.svg';

interface WalletAddTokenCardProps {
  token: IWalletTokenCard;
}

const WalletAddTokenCard: React.FunctionComponent<WalletAddTokenCardProps> = (
  props: WalletAddTokenCardProps
) => {
  const { token } = props;
  return (
    <Card className={styles.tokenCard}>
      <CardBody className={styles.cardBody}>
        <Row>
          <Col md='6'>
            <div className='d-flex'>
              {/* <img src={Icon} /> */}
              <div className='ml-4'>
                <div>
                  <b>{token.symbol}</b>
                </div>
                <div className={styles.cardValue}>{token.name}</div>
              </div>
            </div>
          </Col>
          <Col md='6'>
            <div className='float-right'>
              <Button color='link'>
                <span className='d-lg-inline'>
                  {I18n.t('containers.wallet.walletAddTokensPage.add')}
                </span>
              </Button>
            </div>
          </Col>
        </Row>
      </CardBody>
    </Card>
  );
};

export default WalletAddTokenCard;
