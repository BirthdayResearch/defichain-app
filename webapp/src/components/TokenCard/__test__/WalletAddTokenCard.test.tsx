import React from 'react';
import WalletAddTokenCard from '../WalletAddTokenCard';
import { shallow } from 'enzyme';

describe('WalletAddTokenCard component', () => {
  it('should check for snapshot', () => {
    const IWalletTokenCard = {
      symbolKey: 'symbolKey',
      hash: 'hash',
      name: 'name',
      symbol: 'symbol',
      isDAT: true,
      isLPS: true,
      decimal: 1,
      limit: 1,
      mintable: true,
      tradeable: true,
      creationTx: 'creationTx',
      creationHeight: 2,
      destructionTx: 'destructionTx',
      destructionHeight: 1,
      amount: 1,
      address: 'address',
    };
    const props = {
      token: IWalletTokenCard,
    };
    const wrapper = shallow(<WalletAddTokenCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
