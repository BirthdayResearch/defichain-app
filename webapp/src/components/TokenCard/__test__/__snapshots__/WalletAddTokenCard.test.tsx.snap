// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletAddTokenCard component should check for snapshot 1`] = `
<Card
  tag="div"
>
  <CardBody
    tag="div"
  >
    <Row
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Col
        md="6"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="d-flex"
        >
          <div
            className="ml-4"
          >
            <div>
              <b>
                symbol
              </b>
            </div>
            <div>
              name
            </div>
          </div>
        </div>
      </Col>
      <Col
        md="6"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="float-right"
        >
          <Button
            color="link"
            tag="button"
          >
            <span
              className="d-lg-inline"
            >
              Add
            </span>
          </Button>
        </div>
      </Col>
    </Row>
  </CardBody>
</Card>
`;
